src/main/java/io/cdap/plugin/snowflake/sink/batch/SnowflakeSinkAccessor.java

```diff
@@ -55,14 +65,32 @@ public void uploadStream(InputStream inputStream, String stageDir) throws IOExce
                                                                 null,
                                                                 inputStream, filename, true);
     } catch (SQLException e) {
-      throw new IOException(e);
+      String errorMessage = String.format(
+        "Unable to compress and upload data to destination stage '%s'. Filename: '%s'. Error: %s",
+        stageDir, filename, e.getMessage()
+      );
+      List<ErrorCode> errorCodes = Arrays.stream(ErrorCode.values())
+        .filter(errorCode -> errorCode.getSqlState().equals(e.getSQLState())).collect(Collectors.toList());
+      throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+        errorMessage, String.format("Error message: %s", errorMessage),
+        errorCodes.isEmpty() ? ErrorType.UNKNOWN : SnowflakeErrorType.getErrorType(errorCodes.get(0).getMessageCode()),
+        true, ErrorCodeType.SQLSTATE, e.getSQLState(),
+        "https://docs.snowflake.com/en/user-guide/client-connectivity-troubleshooting/error-messages", e);
```

Comment:
There are couple of things, `errorReason` & `errorMessage` in `ProgramFailureException`:

```
errorReason = String.format("Unable to compress '%s' and upload data to destination stage '%s', filename, stageDir);
String supportedDocUrl = "https://docs.snowflake.com/en/user-guide/client-connectivity-troubleshooting/error-messages";
errorMessage = String.format("Unable to compress '%s' and upload data to destination stage '%s' with errorCode: '%s' and sqlState: '%s'.  For more details, see %s", filename, stageDir, e.getErrorCode(), e.getSQLState(), supportedDocUrl);

throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
  errorReason, errorMessage,
  errorCodes.isEmpty() ? ErrorType.UNKNOWN : SnowflakeErrorType.getErrorType(errorCodes.get(0).getMessageCode()),
  true, ErrorCodeType.SQLSTATE, e.getSQLState(), supportedDocUrl, e);
```

Reply:
updated Sink and Source as per mentioned above